spec/lib/api/v3/repositories/revision_representer_spec.rb

```diff
@@ -0,0 +1,116 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe ::API::V3::Repositories::RevisionRepresenter do
+  include ::API::V3::Utilities::PathHelper
+
+  let(:representer) { described_class.new(revision) }
+
+  let(:project) { FactoryGirl.build :project }
+  let(:repository) { FactoryGirl.build :repository_subversion, project: project }
+  let(:revision) {
+    FactoryGirl.build(:changeset,
+                      id: 42,
+                      revision: '1234',
+                      repository: repository,
+                      comments: commit_message,
+                      committer: 'foo bar <foo@example.org>',
+                      committed_on: DateTime.now,
+                      )
+  }
+
+  let(:commit_message) { 'Some commit message' }
+
+  context 'generation' do
+    subject(:generated) { representer.to_json }
+
+    it { is_expected.to include_json('Revision'.to_json).at_path('_type') }
```

Comment:
I am sure you could find this example somewhere. But more apropriate would be to use `be_json_eql` instead of `include_json`.

`include_json` would also match for

````JSON
{
  "_type": { "noneSense": "Revision", "moreGarbage": 42 }
}
````